### PR TITLE
feat(vungo)[PBJ-1389] add deep copy util

### PR DESCRIPTION
**Pull request by raycai-vungle**
* Add deep copy util to prepare for simplify Copy implementation
*  Gracefully handle nil receiver of Copy
* Add deep copy test util

### CODE REVIEW

File: internal/util/deepcopy.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+// Copiable indicates that an object can be deep copied.
+type Copiable interface {
+	Copy() interface{}
+}
+
+// DeepCopyInt deep copy int. If src is nil, return nil too.
+func DeepCopyInt(src *int) *int {
+	if src == nil {
+		return (*int)(nil)
```
Comment (yingchen0706v) [inline internal/util/deepcopy.go:13]: why we need explicit type conversion here?

Comment (raycai-vungle) [reply internal/util/deepcopy.go:13]: Oh, it's not necessary. I'll remove it.

**Comment by yingchen0706v** [comment]
LGTM

**Comment by Fan-Xu** [comment]
LGTM

**Comment by Fan-Xu** [comment]
LGTM

**Pull request merged**